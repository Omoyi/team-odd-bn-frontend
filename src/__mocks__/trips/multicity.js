const multicityProps = {
  multicityTripData: {
    data: [
      {
        originId: 1,
        destinationId: 2,
        reason: 'I am reason',
        startDate: '2020-02-02',
        returnDate: '2020-04-04',
      },
    ],
    error: '',
    message: 'Trip requested successfully',
  },
  viewProfile: {
    spinner: true,
  },
  addMoreTrip: jest.fn(),
  editMultipleTrip: jest.fn(),
  removeMoreTrip: jest.fn(),
  sendMulticityTrip: jest.fn(),
  updateSpinnerStatus: jest.fn(),
  tripAction: {
    cities: {
      data: ['tripA', 'tripB'],
    },
    tripIndex: 2,
    removeTrip: jest.fn(),
    onHandleInputChange: jest.fn(),
    data: ['A', 'B'],
    itemRequests: {
      originId: 1,
      destinationId: 2,
      reason: '',
      startDate: '',
      returnDate: '',
    },
  },
  getCitiesAction: jest.fn(),
  history: {
    push: jest.fn(),
  },
};

export const checkInputData = {
  payload: {
    name: 'originId',
    value: 2,
  },
  payload2: {
    name: 'destinationId',
    value: 2,
  },
  payload3: {
    fekeName: 'destinationId',
    fekeValue: 2,
  },
};

export const multicityTripAction = {
  editInputForm: {
    type: 'FEKE_ACTION',
    payload: 'I am data',
  },
  addMoreButton: [
    {
      type: 'ADD_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
      payload: {
        originId: '',
        destinationId: '',
        startDate: '',
        returnDate: '',
        reason: '',
      },
    },
  ],
  removeMoreButton: [
    {
      type: 'REMOVE_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
      payload: 2,
    },
  ],
  editMultipleTrip: [
    {
      type: 'EDIT_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
      payload: { name: 'reason', value: 'why' },
    },
  ],
  multicityObjectData: {
    itinerary: [
      {
        originId: '',
        destinationId: '',
        startDate: '',
        returnDate: '',
        reason: '',
      },
    ],
  },
  expectedMulticityReturn: [
    {
      type: 'SEND_MULTIPLE_DATA_SUCCESS',
      payload: { message: 'Created successfully' },
    },
  ],
  currentMulticityRequest: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: [],
  },
  expectedEditedMulticityRequest: [
    {
      originId: 3,
      destinationId: '6',
      startDate: '2020-12-12',
      returnDate: '2021-12-12',
      reason: 'You should try multicity'
    },
    {
      originId: 9,
      destinationId: '2',
      startDate: '2022-12-12',
      returnDate: '2025-12-12',
      reason: 'You should again try multicity',
    },
  ],
  multicityInputDataPayload: {
    name: 'originId',
    value: 9,
    tripIndex: 1,
  },
  addNewTripRequestResult: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
      {
        originId: '',
        destinationId: '',
        startDate: '',
        returnDate: '',
        reason: '',
      },
    ],
    error: [],
  },
  removeTripdReducer: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: [],
  },
  removeReducerAction:
  {
    type: 'REMOVE_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
    payload: 0,
  },
  removedDataNewObject: {
    data: [
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity'
      },
    ],
    error: [],
  },
  editSpecificTripRequest: {
    type: 'EDIT_TRIP_REQUEST_MULTIPLE_DATA_SUCCESS',
    payload: {
      tripIndex: 1,
      value: 'i am reason',
      name: 'reason',
    },
  },
  editSpecificTripRequestReason: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'i am reason',
      },
    ],
    error: [],
  },
  successfullyMessage: {
    type: 'SEND_MULTIPLE_DATA_SUCCESS',
    payload: {
      message: 'successfuly created',
    },
  },
  successfullyTripRequestSaved: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: [],
    message: 'successfuly created',
  },
  errorTripRequestMessage: {
    type: 'SEND_MULTIPLE_DATA_ERROR',
    payload: {
      message: 'server is down',
    },
  },
  erroTripRequestSaved: {
    data: [
      {
        originId: 3,
        destinationId: '6',
        startDate: '2020-12-12',
        returnDate: '2021-12-12',
        reason: 'You should try multicity',
      },
      {
        originId: '1',
        destinationId: '2',
        startDate: '2022-12-12',
        returnDate: '2025-12-12',
        reason: 'You should again try multicity',
      },
    ],
    error: 'server is down',
  },
};


export default multicityProps;
