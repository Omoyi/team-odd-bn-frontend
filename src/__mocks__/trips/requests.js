import React from 'react';
import { Link } from 'react-router-dom';

export const tableHeads = [
  { className: 'tripIdNav', attribute: 'Trip Id', key: 'tripId' },
  { className: 'tripTypeNav', attribute: 'Trip Type', key: 'tripType' },
  { className: 'createdAtNav', attribute: 'Created At', key: 'createdAt' },
  { className: 'status', attribute: 'Status', key: 'status' },
  { className: 'actionNav', attribute: 'Action', key: 'action' },
];

export const navs = [
  {
    className: 'tableHeader',
    attribute:
  <div>
    <h2>My Requests</h2>
    <h1>here</h1>
  </div>,
    key: 'tableHeader',
  },
  {
    className: 'search',
    attribute: <input type="text" placeholder="search by date" />,
    key: 'search',
  },
];
export const entities = [
  [
    { className: 'tripId', attribute: '1', key: 'tripId' },
    { className: 'tripType', attribute: 'one way', key: 'tripType' },
    { className: 'createdAt', attribute: 'today', key: 'createdAt' },
    { className: 'status', attribute: 'pending', key: 'status' },
    { className: 'action', attribute: 'view', key: 'action' },
  ],
];
export const init = {
  allAccomodation: {
    accommodations: [
      {
        id: 46,
        userId: 15,
        name: 'Victor Palace',
        cityId: 5,
        address: 'Lagos',
        description: ' Queens palace is the best accomodation facility you could ever get',
        googleCoordinates: '-77, -56',
        accommodationRooms: [
          {
            id: 90,
            accommodationId: 46,
            name: '12',
            roomType: 'single',
          },
          {
            id: 91,
            accommodationId: 46,
            name: '13',
            roomType: 'single',
          },
          {
            id: 92,
            accommodationId: 46,
            name: '14',
            roomType: 'single',
          },
        ],
        imagesAccommodation: [
          {
            id: 63,
            accommodationId: 46,
            imageUrl: 'https://images.unsplash.com/photo-1549638441-b787d2e11f14?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
          },
        ],
        city: {
          id: 2,
          city: 'Kigali',
        },
        ratings: [
          {
            id: 28,
            userId: 153,
            accommodationId: 46,
            rating: '5',
            review: 'It has wifi issue',
          },
        ],
      }],
  },
  notifications: {
    notifications: {
      allNotifications: [],
    },
  },
  bookings: {
    bookAccommodation: {
      bookAccommodationInput: {
        destination: 'Kigali',
        accommodationId: 45,
        roomType: 'single',
        roomId: 23,
        checkOutDate: '2020-12-11',
        checkInDate: '2022-12-11',
        tripId: 21,
      },
    },
  },
  search: {
    key: '',
    payload: [],
  },
  popUpsDisplay: {
    bookAccommodation: 'none',
    deleteComment: 'none',
    editTrip: 'none',
    currentPopUp: 'editTrip',
  },
  pagination: {
    pageNo: 0,
    itemsPerPage: 7,
  },
  viewProfile: {
    profile: {},
  },
  profileError: { error: 'wrong' },
  auth: {
    signup: {
      signupInputData: {},
      signupData: { blockSpinner: 'blockSpinner' },
      signupError: { blockSpinner: 'blockSpinner' },
    },
    signin: {
      signinInputData: {},
      signinData: { blockSpinner: 'blockSpinner' },
      signinError: { blockSpinner: 'blockSpinner' },
    },
  },
  trips: {
    availRequests: {
      approveRequestMessage: {
      },
    },
    tripRequests: { getCity: [{ id: 1, city: 'kenya' }, { id: 2, city: 'kenya' }] },
    requests: {
      requestsData: {
        data: [
          {
            id: 5,
            userId: 1,
            tripTypeId: 3,
            statusId: 1,
            createdAt: '2020-01-11T12:34:27.264Z',
            updatedAt: '2020-01-11T12:34:27.264Z',
            trips: [
              {
                id: 6,
                tripRequestId: 5,
                originId: 'Nairobi',
                destinationId: 'Kigali',
                reason: 'kimjong',
                startDate: '2029-12-23',
                returnDate: null,
                createdAt: '2020-01-11T12:34:27.329Z',
                updatedAt: '2020-01-11T12:34:27.329Z',
              },
            ],
            tripType: {
              id: 3,
              tripType: 'MultiCity',
              createdAt: '2019-12-12T19:12:16.590Z',
              updatedAt: '2019-12-12T19:12:16.590Z',
            },
            status: {
              id: 1,
              status: 'Pending',
              createdAt: '2019-12-12T19:12:16.595Z',
              updatedAt: '2019-12-12T19:12:16.595Z',
            },
          },
        ],
      },
      singleRequestData: {
        data: {
          id: 8,
          userId: 6,
          tripTypeId: 3,
          statusId: 1,
          createdAt: '2020-01-11T22:08:54.891Z',
          updatedAt: '2020-01-11T22:08:54.891Z',
          tripType: {
            id: 3,
            tripType: 'MultiCity',
            createdAt: '2019-12-12T19:12:16.590Z',
            updatedAt: '2019-12-12T19:12:16.590Z',
          },
          status: {
            id: 1,
            status: 'Pending',
            createdAt: '2019-12-12T19:12:16.595Z',
            updatedAt: '2019-12-12T19:12:16.595Z',
          },
          user: {
            firstName: 'john',
            lastName: 'doe',
            id: 6,
          },
          trips: [
            {
              id: 10,
              tripRequestId: 8,
              originId: 1,
              destinationId: 2,
              reason: 'kimjong',
              startDate: '2069-12-23',
              returnDate: null,
              createdAt: '2020-01-11T22:08:54.908Z',
              updatedAt: '2020-01-11T22:08:54.908Z',
            },
          ],
        },
      },

      requestCommentsData: [
        {
          id: 3,
          comment: 'commejfgjhkhhgnt',
          updatedAt: '2020-01-12T07:41:14.461Z',
          user: {
            firstName: 'John',
            lastName: 'Doe',
            userProfile: {
              imageURL: '/images/me.jpg',
            },
          },
        },
      ],
      requestsError: {},
    },
  },
};
export const requestViewData = [
  {
    id: 1,
    tripType: { tripType: 'one way' },
    createdAt: '2011-30-24',
    status: { status: 'pending' },
  },
];
export const ManagerViewData = [
  {
    user: { firstName: 'Ivy', lastName: 'League' },
    tripType: { tripType: 'one way' },
    createdAt: '2011-30-24',
  },
];
