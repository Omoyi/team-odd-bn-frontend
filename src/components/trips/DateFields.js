import React from 'react';
import '../../assets/css/base.scss';
import '../../assets/css/roundtrip.scss';

const inputValues = {};
const checkCurrentDate = () => {
  const currentDate = new Date();
  return JSON.stringify(currentDate).slice(1, 11);
};
const DateFields = ({ dateFieldName, handleChange }) => dateFieldName.map((x) => (

  <div className={x.className} key={x.className}>
    <label htmlFor={x.htmlForName}>{ x.FieldName }</label>
    <input
      data_test={x.data_test}
      type={x.type}
      name={x.name}
      required
      min={checkCurrentDate()}
      id={x.id}
      value={x.value}
      onChange={(e) => {
        inputValues[x.name] = e.target.value;
        return handleChange(inputValues);
      }}
    />
  </div>

));

export default DateFields;
