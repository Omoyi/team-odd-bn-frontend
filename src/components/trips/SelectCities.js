import React from 'react';
import '../../assets/css/base.scss';

const inputsValues = {};
const SelectCities = ({ myData, handleChange, inputName }) => inputName.map((a) => (
  <div className={a.className} key={a.className}>
    <select
      onChange={(e) => {
        inputsValues[a.name] = e.target.value;
        return handleChange(inputsValues);
      }}
      defaultValue=""
      required
    >
      <option value="" disabled>
        { a.PlaceHolderName}
      </option>

      {myData
      && myData.map((cities) => (
        <option name={a.name} value={cities.id} key={cities.id}>
          {cities.city}
        </option>
      ))}
    </select>
  </div>
));

export default SelectCities;
