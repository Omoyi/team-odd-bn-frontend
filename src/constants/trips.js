import React from 'react';
import { Link } from 'react-router-dom';
import StatsNav from '../components/GetStatsNav';

export const tableHeads = [
  { className: 'tripTypeNav', attribute: 'Trip Type', key: 'tripType' },
  { className: 'status', attribute: 'Status', key: 'status' },
  { className: 'createdAtNav', attribute: 'Requested On', key: 'Requested On' },
  { className: 'actionNav', attribute: 'Action', key: 'action' },
];
export const navs = [
  {
    className: 'tableHeader',
    attribute:
  <div className="navRequests">
    <div className="travelRequestHeaders"><h2>My Travel Requests</h2></div>
    <div className="newTrip">
      <Link className="add-newTrip-Link" to="/trips/oneway">Add New Trip</Link>
    </div>
  </div>,
    key: 'tableHeader',
  },

];

export const approvalNavs = [
  { className: 'tableHeader', attribute: <h2>Travel Requests For Approval</h2>, key: 'tableHeader' },

];
export const approvalTableHeads = [
  { attribute: 'First Name', key: 'firstname' },
  { attribute: 'Last Name', key: 'lastname' },
  { attribute: 'Trip Type', key: 'tripType' },
  { attribute: 'Requested On', key: 'requested-on' },
  { attribute: 'Action', key: 'action' },
];

export const searchTableHeads = [

  { className: 'tripTypeNav', attribute: 'Origin', key: 'tripType' },
  { className: 'createdAtNav', attribute: 'Destination', key: 'createdAt' },
  { className: 'status', attribute: 'Date', key: 'status' },
  { className: 'actionNav', attribute: 'Action', key: 'action' },
];
export const searchNavs = [
  { className: 'tableHeader', attribute: <StatsNav />, key: 'tableHeader' },
];
export const newTrip = {
  originId: '',
  destinationId: '',
  reason: '',
  startDate: '',
  returnDate: '',
};
