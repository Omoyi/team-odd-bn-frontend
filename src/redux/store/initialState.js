const state = {
  accommodation: {
    createAccommodation: {
      rooms: [],
      photos: [],
    },
  },
  auth: {
    signup: {
      spinnerStatus: false,
      signupInputData: {},
      signupData: { },
      signupError: { },
    },
    signin: {
      spinnerStatus: false,
      signinInputData: {},
      signinData: {},
      signinError: {},
    },
  },
  trips: {
    tripRequests: {
      getCity: [],
      onewayData: {},
      onewayError: {},
      onewayInput: {},
    },
    availRequests: {
      approveRequestMessage: {
      },
    },
    requests: {
      requestsData: [],
      singleRequestData: {},
      requestCommentsData: {},
      requestsError: {},
    },
  },
  viewProfile: {
    profile: {},
  },
};
export default state;
