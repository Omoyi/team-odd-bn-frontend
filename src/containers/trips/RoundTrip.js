import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../../assets/css/roundtrip.scss';
import { Redirect } from 'react-router';
import Submit from '../../components/SubmitButton';
import Dashboard from '../../views/Dashboard/sidebar/index';
import '../../assets/css/base.scss';
import TripType from '../../components/trips/TripTypes';
import TripHeader from '../../components/trips/TripHeader';
import createRoundTrip, { updateSpinnerStatus } from '../../redux/actions/roundTripActions';
import SelectCities from '../../components/trips/SelectCities';
import DateFields from '../../components/trips/DateFields';
import selectInputs from '../../constants/selectCities';
import {
  getTripAction,
} from '../../redux/actions/tripsActions/onewayActions';
import DateFieldInput from '../../constants/DateFieldInput';

export class RoundTrip extends Component {
  constructor(props) {
    super(props);
    this.state = {
      originId: '',
      destinationId: '',
      reason: '',
      startDate: '',
      returnDate: '',
      formErrors: {
        reason: '',
        startDate: '',
        returnDate: '',
      },
    };

    this.handleChangeInput = this.handleChangeInput.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  async componentDidMount() {
    const { createRoundTrip, updateSpinnerStatus, getTripAction } = this.props;
    await updateSpinnerStatus();
    await getTripAction();
  }

  handleChangeInput = data => {
    this.setState({ ...data, data })
  }

  async handleSubmit(e) {
    e.preventDefault();
    updateSpinnerStatus(true);

    const {
      originId,
      destinationId,
      reason,
      startDate,
      returnDate,
    } = this.state;
    
    const data = {
      originId: parseInt(originId, 10),
      destinationId: parseInt(destinationId, 10),
      reason,
      startDate: startDate,
      returnDate: returnDate,
    };
    await this.props.createRoundTrip(data);
  }

  render() {
    const {
      spinner, roundTripMessage, roundTripError, cities, tripState,
    } = this.props;

    const {
      reason,
    } = this.state;

    const data = tripState.trips.tripRequests.getCity;    
    return (

      <Dashboard>
        { roundTripMessage && <Redirect to={`/requests/`} />}
        <div className="oneway-container">


          <TripHeader />
          <div className="trip-field-container">
            <TripType />
            <div className="trip-form">
              <form
                onSubmit={this.handleSubmit}
              >
                <div className="cityField">
                  <div className="cityLabel">
                    <label htmlFor="input">City</label>
                  </div>
                  <SelectCities handleChange={this.handleChangeInput} inputName={selectInputs} myData={data} />
                </div>

                <div className="reasonField">
                  <label htmlFor="textArea">Reason</label>
                  <textarea
                    data_test="reason"
                    id="textArea"
                    cols="30"
                    rows="10"
                    onChange={e => this.handleChangeInput({reason: e.target.value})}
                    value={reason}
                    required
                    min='3'
                    name="reason"
                    placeholder="Reason for trip request"
                  />
                </div>

                <div className="dateField">
                  <DateFields dateFieldName={DateFieldInput}  handleChange={this.handleChangeInput}/>
                </div>

                <div className="btnField">
                  <Submit buttonName="Request trip" className="btn-submit" />
                </div>
              </form>
            </div>
          </div>
        </div>
      </Dashboard>
    );
  }
}

export const mapStateToProps = (state) => ({
  tripState: state,
  roundTripMessage: state.trips.roundTrip.roundTripMessage,
  roundTripError: state.trips.roundTrip.roundTripError,
  spinner: state.trips.roundTrip.spinner,
  spinnerError: state.trips.roundTrip.spinnerError,
  cities: state.allCities.cities,
  citiesError: state.allCities.citiesError,

});

export default connect(mapStateToProps, { createRoundTrip, updateSpinnerStatus, getTripAction })(RoundTrip);
