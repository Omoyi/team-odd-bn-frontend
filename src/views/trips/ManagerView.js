import React from 'react';
import { connect } from 'react-redux';
import CommonTable from './CommonTable';
import Action from '../../components/Action';
import { approvalNavs, approvalTableHeads } from '../../constants/trips';
import { Paginate } from '../../helpers/Paginate';

const AvailRequestsView = ({ availRequests, pageNo, itemsPerPage }) => {
  const paginatedData = availRequests && availRequests.data;

  const entities = paginatedData
  && { ...Paginate(paginatedData, itemsPerPage) }[pageNo].map((request) => [
    { attribute: request.user && request.user.firstName, key: 'firstName' },
    { attribute: request.user && request.user.lastName, key: 'lastName' },
    // { attribute: request.tripType.tripType, key: 'tripType' },
    // { attribute: request.createdAt.slice(0, 10), key: 'created-at' },
    { attribute: <Action background="#34c6f3" url={`/trips/approval/${request.id}`} action="View" />, key: 'action' },
  ]);

  return (
    <CommonTable
      data={paginatedData && Paginate(paginatedData, itemsPerPage)}
      navs={approvalNavs}
      tableHeads={approvalTableHeads}
      entities={entities}
    />
  );
};


export const mapStateToProps = (state) => ({
  availRequests: state.trips.availRequests.allRequests,
  pageNo: state.pagination.pageNo,
  itemsPerPage: state.pagination.itemsPerPage,
});


export default connect(mapStateToProps, null)(AvailRequestsView);
