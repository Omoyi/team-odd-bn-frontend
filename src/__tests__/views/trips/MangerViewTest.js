import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ManagerView from '../../../views/trips/ManagerView';
import { ManagerViewData } from '../../../__mocks__/trips/requests';
import initialState from '../../../redux/store/initialState';

let wrapper;
const mockStore = configureStore([]);
let store;
describe('Manager View Test Suite ', () => {
  store = mockStore(initialState);

  it('Test snapshot using the initial state', () => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <ManagerView data={ManagerViewData} />
        </Router>
      </Provider>,

    );
    expect(wrapper).toMatchSnapshot();
  });
});
