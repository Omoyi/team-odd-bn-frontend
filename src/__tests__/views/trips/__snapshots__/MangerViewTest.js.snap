// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manager View Test Suite  Test snapshot using the initial state 1`] = `ReactWrapper {}`;
